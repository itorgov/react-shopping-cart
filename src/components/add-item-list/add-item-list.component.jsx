import React from 'react';

import {AddItem} from '../add-item/add-item.component';

import styles from './add-item-list.module.scss';

export const AddItemList = ({items}) => (
    <div className={styles.list}>
        {items.map(item => <AddItem key={item.id} item={item}/> )}
    </div>
);
