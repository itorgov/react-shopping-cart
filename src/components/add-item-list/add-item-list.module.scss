.list {
  padding: 3rem;

  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-gap: 6rem;

  @media only screen and (max-width: 37.5em) {
    grid-template-columns: repeat(2, 1fr);
    grid-gap: 3rem;
  }

  @media only screen and (max-width: 31.25em) {
    grid-template-columns: 1fr;
    grid-row-gap: 6rem;
  }
}
