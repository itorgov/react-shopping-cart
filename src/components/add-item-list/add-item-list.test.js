import React from "react";
import {shallow} from "enzyme";

import {AddItemList} from './add-item-list.component';

describe('<AddItemList />', () => {
    it('expect to render', () => {
        const mockItems = [
            {
                id: 1,
                name: "Test name A",
                price: 4000,
            },
            {
                id: 1,
                name: "Test name B",
                price: 2000,
            },
        ];
        expect(shallow(<AddItemList items={mockItems} />)).toMatchSnapshot();
    });
});
