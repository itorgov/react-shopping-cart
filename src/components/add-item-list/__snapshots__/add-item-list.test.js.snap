// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddItemList /> expect to render 1`] = `
<div
  className="list"
>
  <AddItem
    item={
      Object {
        "id": 1,
        "name": "Test name A",
        "price": 4000,
      }
    }
    key="1"
  />
  <AddItem
    item={
      Object {
        "id": 1,
        "name": "Test name B",
        "price": 2000,
      }
    }
    key="1"
  />
</div>
`;
